File: color-spaces/mask/mask.py
# -----------------------------------------------------------------------------
#
# Example for highlight blue shades from RGB on Python3 with OpenCV.
#
# -----------------------------------------------------------------------------

import numpy as np
import cv2

imageRGB = cv2.imread("image.png")

if imageRGB is None:
    exit(-1)

lowerBlue = np.array([15, 15, 0])
upperBlue = np.array([130, 170, 255])

mask = cv2.inRange(imageRGB, lowerBlue, upperBlue)

blueRegions = cv2.bitwise_and(imageRGB, imageRGB, mask=mask)

cv2.imwrite("imageBlue.png", blueRegions)

File: color-spaces/mask/mask.cpp
//-----------------------------------------------------------------------------
//
// Example for highlight blue shades from RGB image on C++ with OpenCV.
//
//-----------------------------------------------------------------------------

#include <opencv2/opencv.hpp>

#include <algorithm>
#include <vector>

int main() {
    cv::Mat imageRGB = cv::imread("image.png");
    cv::Mat blueRegions, imageBlue;

    if (imageRGB.empty()) exit(-1);

    cv::Scalar lowerBlue{15, 15, 0};
    cv::Scalar upperBlue{130, 170, 255};

    cv::inRange(imageRGB, lowerBlue, upperBlue, blueRegions);

    cv::bitwise_and(imageRGB, imageRGB, imageBlue, blueRegions);

    cv::imwrite("imageBlue.png", imageBlue);
}
